app/src/main/java/com/dodo/flashcards/presentation/forgotPassScreen/ForgotPassViewModel.kt

```diff
@@ -32,11 +34,13 @@ class ForgotPassViewModel @Inject constructor(
     private fun onClickedConfirmEmail() {
         viewModelScope.launch(Dispatchers.IO) {
             lastPushedState?.apply {
-                if (forgotPassSendEmailUseCase(textEmail)) {
-                    PendingConfirmation(textEmail)
-                } else {
-                    InvalidEmail(textEmail)
-                }.push()
+                forgotPassSendEmailUseCase(textEmail)
+                    .doOnSuccess {
+                        PendingConfirmation(textEmail).push()
+                    }
+                    .doOnError {
+                        InvalidEmail(textEmail).push()
+                    }
```

Comment:
The other reason I wanted to do this was so we could still access the `.doOnSuccess` & error extensions, seen in this rewrite.